src/photon/photon_algorithm.c

```diff
@@ -301,50 +378,30 @@ void handle_worker_available(local_scheduler_state *state,
   worker *available_worker =
       (worker *) utarray_eltptr(state->workers, worker_index);
   CHECK(available_worker->task_in_progress == NULL);
-  /* Try to schedule another task to the worker. */
-  int scheduled_task =
-      find_and_schedule_task_if_possible(state, algorithm_state, worker_index);
-  /* If we couldn't find a task to schedule, add the worker to the queue of
-   * available workers. */
-  if (!scheduled_task) {
-    for (int *p = (int *) utarray_front(algorithm_state->available_workers);
-         p != NULL;
-         p = (int *) utarray_next(algorithm_state->available_workers, p)) {
-      DCHECK(*p != worker_index);
-    }
-    /* Add client_sock to a list of available workers. This struct will be freed
-     * when a task is assigned to this worker. */
-    utarray_push_back(algorithm_state->available_workers, &worker_index);
-    LOG_DEBUG("Adding worker_index %d to available workers.\n", worker_index);
+  for (int *p = (int *) utarray_front(algorithm_state->available_workers);
+       p != NULL;
+       p = (int *) utarray_next(algorithm_state->available_workers, p)) {
+    DCHECK(*p != worker_index);
   }
+  /* Add worker to the list of available workers. */
+  utarray_push_back(algorithm_state->available_workers, &worker_index);
+  LOG_DEBUG("Adding worker_index %d to available workers.\n", worker_index);
```

Comment:
Doesn't really matter, but there shouldn't be a `\n` in the LOG_DEBUG. (I realize it was already there)